bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AbstractComponentImpl.java

```diff
@@ -58,27 +55,33 @@
     @Nullable
     protected ComponentContext componentContext;
 
+    /**
+     * The current page.
+     */
     @ScriptVariable(injectionStrategy = InjectionStrategy.OPTIONAL)
     @Nullable
     private Page currentPage;
 
+    /**
+     * The ID for this component.
+     */
     private String id;
+
+    /**
+     * Flag indicating if the data layer is enabled.
+     */
     private Boolean dataLayerEnabled;
+
+    /**
+     * The data layer component data.
+     */
     private ComponentData componentData;
 
-    @Nullable
+    @NotNull
     @Override
     public String getId() {
         if (id == null) {
-            if (resource != null) {
-                ValueMap properties = resource.getValueMap();
-                id = properties.get(Component.PN_ID, String.class);
-            }
-            if (StringUtils.isEmpty(id)) {
-                id = generateId();
-            } else {
-                id = StringUtils.replace(StringUtils.normalizeSpace(StringUtils.trim(id)), " ", ID_SEPARATOR);
-            }
+            this.id = DataLayerUtils.getPropertyOrGeneratedId(this.resource, this.currentPage, this.componentContext);
```

Comment:
See my comment below.

Reply:
`DataLayerUtils` has been moved and renamed to `ComponentUtils`.
Will mark this as resolved.